tests/unit/test_kernel_checkpoint.py

```
@@ -0,0 +1,225 @@
+    assert a != b
+
+
+def test_key_is_human_readable():
+    key = compute_checkpoint_key(project_id="proj-123", init_source_hash="abc", environment_id="py311")
+    assert "proj-123" in key
+    assert "abc" in key
+    assert "py311" in key
+
+
+def test_key_rejects_empty_inputs():
+    with pytest.raises(ValueError):
```

Ruff (0.15.14)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)